packages/expo-camera/src/CameraModule/CameraUtils.ts

```diff
@@ -171,3 +178,199 @@ export async function getStreamDevice(
 export function isWebKit(): boolean {
   return /WebKit/.test(navigator.userAgent) && !/Edg/.test(navigator.userAgent);
 }
+
+export function compareStreams(a: MediaStream | null, b: MediaStream | null): boolean {
+  if (!a || !b) return false;
+  const settingsA = a.getTracks()[0].getSettings();
+  const settingsB = b.getTracks()[0].getSettings();
+  return settingsA.deviceId === settingsB.deviceId;
+}
+
+export function capture(
+  video: HTMLVideoElement,
+  settings: MediaTrackSettings,
+  config: CameraPictureOptions
+): CapturedPicture {
+  const base64 = captureImage(video, config);
+
+  const capturedPicture: CapturedPicture = {
+    uri: base64,
+    base64,
+    width: 0,
+    height: 0,
+  };
+
+  if (settings) {
+    const { width = 0, height = 0 } = settings;
+    capturedPicture.width = width;
+    capturedPicture.height = height;
+    // TODO: Bacon: verify/enforce exif shape.
+    capturedPicture.exif = settings;
+  }
+
+  if (config.onPictureSaved) {
+    config.onPictureSaved({ nativeEvent: { data: capturedPicture, id: config.id } });
+  }
+  return capturedPicture;
+}
+
+export async function syncTrackCapabilities(
+  cameraType: CameraType,
+  stream: MediaStream | null,
+  settings: WebCameraSettings = {}
+): Promise<void> {
+  if (stream?.getVideoTracks) {
+    await Promise.all(
+      stream.getVideoTracks().map(track => onCapabilitiesReady(cameraType, track, settings))
+    );
+  }
+}
+
+// https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints
+async function onCapabilitiesReady(
+  cameraType: CameraType,
+  track: MediaStreamTrack,
+  settings: WebCameraSettings = {}
+): Promise<void> {
+  console.log('sync: ', cameraType, settings);
+  const capabilities = track.getCapabilities();
+
+  // Create an empty object because if you set a constraint that isn't available an error will be thrown.
+  const constraints: MediaTrackConstraintSet = {};
+
+  // TODO: Bacon: Add `pointsOfInterest` support
+  const clampedValues = [
+    'exposureCompensation',
+    'colorTemperature',
+    'iso',
+    'brightness',
+    'contrast',
+    'saturation',
+    'sharpness',
+    'focusDistance',
+    'zoom',
+  ];
+
+  for (const property of clampedValues) {
+    if (capabilities[property]) {
+      constraints[property] = convertNormalizedSetting(capabilities[property], settings[property]);
+    }
+  }
+
+  const _validatedConstrainedValue = (key, propName, converter) =>
```

Comment:
```suggestion
  const _validatedConstrainedValue = (key: string, propName: string, converter: (value: any) => any) =>
```
I'm not a fun of these `any` in `converter` 😞